tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -896,17 +898,53 @@ def return_it(x):
 @cocotb.test()
 def test_exceptions(dut):
     @cocotb.coroutine
-    def raise_soon():
+    def raise_inner():
         yield Timer(10)
         raise ValueError('It is soon now')
 
+
+    @cocotb.coroutine
+    def raise_soon():
+        yield Timer(1)
+        yield raise_inner()
+
     try:
         yield raise_soon()
     except ValueError:
-        pass
+        tb_text = traceback.format_exc()
     else:
         raise TestFailure("Exception was not raised")
 
+    # check the traceback is readable
+    expected = textwrap.dedent(r"""
+    Traceback \(most recent call last\):
+      File ".*test_cocotb\.py", line \d+, in test_exceptions
+        yield raise_soon\(\)
+      File ".*decorators\.py", line \d+, in _advance
+        .*
+      File ".*outcomes\.py", line \d+, in send
+        .*
+      File ".*test_cocotb\.py", line \d+, in raise_soon
+        yield raise_inner\(\)
+      File ".*decorators\.py", line \d+, in _advance
+        .*
+      File ".*outcomes\.py", line \d+, in send
+        .*
+      File ".*test_cocotb\.py", line \d+, in raise_inner
+        raise ValueError\('It is soon now'\)
+    ValueError: It is soon now""").strip()
+
+
+    if not re.match(expected, tb_text):
+        raise TestFailure(
+            (
+                "Traceback didn't match - got:\n"
```

Comment:
Pretty sure this will fail in CI - will push a new rev once I have results to go off